Makefile

```diff
@@ -233,6 +224,13 @@ endif
 endif
 	$(tools_bin_path)/htmltest -c $(BUILD)/.htmltest.yml
 
+.PHONY: lint
+lint: export GO_BUILD_TAGS = $(GO_BUILD_TAGS_LIST)
+lint: install-tools linter
+	@echo "--- :golang: Running linters"
+	./scripts/run-ci-lint.sh $(tools_bin_path)/golangci-lint
+	./bin/linter ./...
```

Comment:
Can we make this able to run by subdir?

Reply:
Good point, will do.

Reply:
late to the party, but running by subdir multiplies the runtime by # of subdirs you have, and defeats most of golangci-lint optimizations (only parsing code once).
this should not be a default mode of operation.

Reply:
Addressed in https://github.com/m3db/m3/pull/2863